src/Builders/ComponentParameterBuilder.cs

```diff
@@ -0,0 +1,351 @@
+        {
+            if (parameterSelector is null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+
+            if (markup is null)
+            {
+                throw new ArgumentNullException(nameof(markup));
+            }
+
+            var (name, isCascading) = GetDetailsFromExpression(parameterSelector);
+            return AddParameterToList(name, markup.ToMarkupRenderFragment(), isCascading);
+        }
+
+        /// <summary>
+        /// Add a strongly typed <see cref="RenderFragment{TValue}" /> parameter with a template for a component under test.
+        /// </summary>
+        /// <typeparam name="TValue">The generic value type</typeparam>
+        /// <param name="parameterSelector">The parameter selector which defines the parameter to add</param>
+        /// <param name="template"><see cref="RenderFragment{TValue}" /> to pass to the parameter</param>
+        /// <returns>A <see cref="ComponentParameterBuilder{TComponent}"/> which can be chained</returns>
+        public ComponentParameterBuilder<TComponent> Add<TValue>(Expression<Func<TComponent, RenderFragment<TValue>?>> parameterSelector, RenderFragment<TValue> template)
+        {
+            if (parameterSelector is null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+
+            if (template is null)
+            {
+                throw new ArgumentNullException(nameof(template));
+            }
+
+            var (name, isCascading) = GetDetailsFromExpression(parameterSelector);
+            return AddParameterToList(name, template, isCascading);
+        }
+
+        /// <summary>
+        /// Add a strongly typed <see cref="RenderFragment{TValue}" /> parameter with a markupFactory for a component under test.
+        /// </summary>
+        /// <typeparam name="TValue">The generic value type</typeparam>
+        /// <param name="parameterSelector">The parameter selector which defines the parameter to add</param>
+        /// <param name="markupFactory">A markup factory that takes a <typeparamref name="TValue"/> as input and returns markup/HTML.</param>
+        /// <returns>A <see cref="ComponentParameterBuilder{TComponent}"/> which can be chained</returns>
+        public ComponentParameterBuilder<TComponent> Add<TValue>(Expression<Func<TComponent, RenderFragment<TValue>?>> parameterSelector, Func<TValue, string> markupFactory)
+        {
+            if (parameterSelector is null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+
+            if (markupFactory is null)
+            {
+                throw new ArgumentNullException(nameof(markupFactory));
+            }
+
+            return Add(parameterSelector, value => renderTreeBuilder => renderTreeBuilder.AddMarkupContent(0, markupFactory(value)));
+        }

```

Comment:
```suggestion
        public ComponentParameterBuilder<TComponent> Add<TValue>(Expression<Func<TComponent, RenderFragment<TValue>?>> parameterSelector, Func<TValue, string> markupFactory) => Add(parameterSelector, value => renderTreeBuilder => renderTreeBuilder.AddMarkupContent(0, markupFactory(value)));
```
I think this should be OK.